charts/tekton-apps/templates/wordpress/argocd_applications.yaml

```diff
@@ -22,161 +17,171 @@ metadata:
-          enabled: true
-          runAsUser: 1000
+          {{- toYaml .containerSecurityContext | nindent 10 }}
         {{- end }}
 
-        {{ if ternary $component.wordpress.ci "true" (hasKey $component.wordpress "ci") }}
+        {{ if or .ci (not (hasKey . "ci")) -}}
         initContainers:
           - name: install
-            image: jnewland/git-and-stuff
+            image: bitnami/git
             imagePullPolicy: Always
+            securityContext:
+              runAsUser: {{ (.containerSecurityContext).runAsUser | default 1001 }}
+            env:
+              - name: HOME
+                value: /home/git
             command:
               - bash
               - -c
               - |
-                git -c core.sshCommand="ssh -i ~/.ssh/id_rsa" clone {{ $component.wordpress.repository_ssh_url }} /tmp/wordpress
-                cp -Rf /tmp/wordpress/wp-content/* /bitnami/wordpress/wp-content/
-                cd /tmp/wordpress
-                git_branch=`git rev-parse --abbrev-ref HEAD`
-                git_log=`git log --format="%h - by %an - %ae - %s" HEAD~1..HEAD`
-                msg="[$git_branch] $git_log"
-                printf "%0.s-" $(seq 1 ${#msg})
-                echo
-                echo "$msg"
-                printf "%0.s-" $(seq 1 ${#msg})
-                echo
-                ls -la /tmp/wordpress/wp-content
-                echo "Folder wp-content copied from git repo into bitnami/wodpress"
-                echo "Done INIT"
+                set -o errexit
+                set -o nounset
+                set -o pipefail
+                #set -o xtrace
+                WP_ROOT=/bitnami/wordpress
+                COPY_WP_CONF="{{ and .repositoryUseWPConfig (not .existingWordPressConfigurationSecret) | default "false"}}"
+                URL="{{ .repositorySshUrl | required "Missed .wordpress.repositorySshUrl option"}}"
+                REVISION="{{ .repositoryRevision | default "" }}"
+                #--------------------------------------------------------------------
+                # chart does not start wp setup if WP_ROOT is not empty
+                if [[ ! -d $WP_ROOT/wp-content && $COPY_WP_CONF == "false" ]]; then
+                  echo "WP not initialized, skipping"
+                else
+                #--------------------------------------------------------------------
+                #setup git user with runAsUser id
+                  if ! getent passwd "$(id -u)" &>/dev/null && [ -e "$NSS_WRAPPER_LIB" ]; then
+                      export LD_PRELOAD="$NSS_WRAPPER_LIB"
+                      # shellcheck disable=SC2155
+                      export NSS_WRAPPER_PASSWD="$(mktemp)"
+                      # shellcheck disable=SC2155
+                      export NSS_WRAPPER_GROUP="$(mktemp)"
+                      echo "git:x:$(id -u):$(id -g):Git:${HOME}:/bin/false" >"$NSS_WRAPPER_PASSWD"
+                      echo "git:x:$(id -g):" >"$NSS_WRAPPER_GROUP"
+                  fi
```

Comment:
why do we need this?

Reply:
use `jnewland/git-and-stuff`

Reply:
I don't understand what value your new implementation adds vs previous implementation. It became substantially longer and the only change is just wp-config copy.

Pls reuse older implementation and add wp-config copy as needed. Make code short like it was in the previous version of the chart.

Reply:
`jnewland/git-and-stuff` binded to 1000 user id, bitnami wordpress container had been setup to 1001 and runAsUser does not works well. It comes to situation where bitnami internal scripts can't proceed with files that they do not own (chown require root rights) and failes on init stage.

For example, before we copied only wp-content files but not folder itself. That works well.
But if we start replace whole wp-content folder - it fails with error in init stage. Same with wp-config.php file

This solution resolves that issue couse init container will have same ids as whole pod.

Reply:
@MikhailKorol-saritasa then set it to 1001 via init container settings. Restore the original code and adjust it properly.

Reply:
Done